realsense2_camera/src/base_realsense_node.cpp

```diff
@@ -495,19 +514,35 @@ void BaseRealSenseNode::imu_callback(rs2::frame frame)
         ImuMessage_AddDefaultValues(imu_msg);
         imu_msg.header.frame_id = OPTICAL_FRAME_ID(stream_index);
 
-        auto crnt_reading = *(reinterpret_cast<const float3*>(frame.get_data()));
-        if (GYRO == stream_index)
+        const float3 *crnt_reading = reinterpret_cast<const float3 *>(frame.get_data());
+        if (IMU == stream_index)
+        {
+            // Expecting two float3 objects: first for accel, second for gyro
+            const float3 &accel_data = crnt_reading[0];
```

Comment:
Can we double check the size before we dereference it?
Otherwise we may crash on SEGFAULT
is it possible to cast to float[] instead of float *? (not sure)

Reply:
added a check for that